tests/components/tts/test_init.py

```diff
@@ -38,612 +76,580 @@ def mutagen_mock():
-    def test_setup_component_demo_no_access_cache_folder(self, mock_mkdir):
-        """Set up the demo platform with defaults."""
-        config = {tts.DOMAIN: {"platform": "demo"}}
+async def test_setup_component_demo_no_access_cache_folder(hass, mock_init_cache_dir):
+    """Set up the demo platform with defaults."""
+    config = {tts.DOMAIN: {"platform": "demo"}}
 
-        assert not setup_component(self.hass, tts.DOMAIN, config)
+    mock_init_cache_dir.side_effect = OSError(2, "No access")
+    assert not await async_setup_component(hass, tts.DOMAIN, config)
 
-        assert not self.hass.services.has_service(tts.DOMAIN, "demo_say")
-        assert not self.hass.services.has_service(tts.DOMAIN, "clear_cache")
+    assert not hass.services.has_service(tts.DOMAIN, "demo_say")
+    assert not hass.services.has_service(tts.DOMAIN, "clear_cache")
 
-    def test_setup_component_and_test_service(self):
-        """Set up the demo platform and call service."""
-        calls = mock_service(self.hass, DOMAIN_MP, SERVICE_PLAY_MEDIA)
 
-        config = {tts.DOMAIN: {"platform": "demo"}}
+async def test_setup_component_and_test_service(hass, empty_cache_dir):
+    """Set up the demo platform and call service."""
+    calls = async_mock_service(hass, DOMAIN_MP, SERVICE_PLAY_MEDIA)
 
-        with assert_setup_component(1, tts.DOMAIN):
-            setup_component(self.hass, tts.DOMAIN, config)
+    config = {tts.DOMAIN: {"platform": "demo"}}
 
-        self.hass.services.call(
-            tts.DOMAIN,
-            "demo_say",
-            {
-                "entity_id": "media_player.something",
-                tts.ATTR_MESSAGE: "There is someone at the door.",
-            },
-        )
-        self.hass.block_till_done()
+    with assert_setup_component(1, tts.DOMAIN):
+        assert await async_setup_component(hass, tts.DOMAIN, config)
+
+    await hass.services.async_call(
+        tts.DOMAIN,
+        "demo_say",
+        {
+            "entity_id": "media_player.something",
+            tts.ATTR_MESSAGE: "There is someone at the door.",
+        },
+        blocking=True,
+    )
 
-        assert len(calls) == 1
-        assert calls[0].data[ATTR_MEDIA_CONTENT_TYPE] == MEDIA_TYPE_MUSIC
-        assert calls[0].data[
-            ATTR_MEDIA_CONTENT_ID
-        ] == "{}/api/tts_proxy/42f18378fd4393d18c8dd11d03fa9563c1e54491_en_-_demo.mp3".format(
-            self.hass.config.api.base_url
-        )
-        assert os.path.isfile(
-            os.path.join(
-                self.default_tts_cache,
-                "42f18378fd4393d18c8dd11d03fa9563c1e54491_en_-_demo.mp3",
-            )
+    assert len(calls) == 1
+    assert calls[0].data[ATTR_MEDIA_CONTENT_TYPE] == MEDIA_TYPE_MUSIC
+    assert calls[0].data[
+        ATTR_MEDIA_CONTENT_ID
+    ] == "{}/api/tts_proxy/42f18378fd4393d18c8dd11d03fa9563c1e54491_en_-_demo.mp3".format(
+        hass.config.api.base_url
+    )
+    assert os.path.isfile(
```

Comment:
If we use the `tmp_path` fixture above we can use the pathlib api here which is nicer than `os.path`.